cmd/config-env.go

```diff
@@ -0,0 +1,590 @@
+package cmd
+
+import (
+	"errors"
+	"os"
+	"strconv"
+)
+
+type envConfigV19 struct {
```

Comment:
We could populate values from environment variables to struct `serverConfigV19`.  No new type is required.

Reply:
@balamurugana values of what structure?

Reply:
@zerkms Actually I am trying to say we would need to add new function in `config-v19.go` something like
```go
func loadConfigEnv() error {
    // Create new `serverConfigV19{}` and fill each fields from the ENV.
    //  If any non-optional field is missing or has wrong value, return error.
}
```

In `server-main.go` we call this funcion first, on error, read config file.  new type `envConfigV19` is not necessary.

Reply:
@balamurugana and as I mentioned before - it won't work: the current implementation (the one before me) relies on the knowledge whether a value came from the environment or from the config file.

Reply:
> So you MUST keep track of environment variables somewhere somehow. At the moment it is implemented as bunch of global variables.

Yes. There are three flags (`globalIsEnvBrowser`, `globalIsEnvCreds` and `globalIsEnvRegion`) we have now to know whether those values are from env vars.

> I refactored it to a single type and a single global variable, which is presumably more clearer.

Could you do this change as separate PR (Previously I was attempting to have and removing global variables led to 3000 lines change) ?  We encourage to have multiple small PRs for easy reviewing.

Also you are welcome to minio.slack.com where we can have longer discussions.

Reply:
@balamurugana

> Could you do this change as separate PR

I though about it initially, but then I would have to reimplement the whole thing twice (given that I do like the solution with a new separated type for environment variables).

Anyway, if you think 

```go
func loadConfigEnv() error {
    // Create new `serverConfigV19{}` and fill each fields from the ENV.
    //  If any non-optional field is missing or has wrong value, return error.
}
```

would work I might invest some of my time more to reimplement it, but I personally don't like it at all:

1. It mutates the global variables (hence it's harder to test)
2. It is validated separately (in my implementation the validation is done just once in one place): maintaining something twice is twice as more error prone than doing something once
3. Because of separate validation - you won't be able to override just some config values but have to provide a full "block" of environment variables: so it makes it much less flexible in use.

PS: I'm in the slack :-)